src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -131,14 +117,49 @@ class wrap {
                 auth_result.pass();
                 break;
             }
+        } catch (const unauthorized_user_exception& e) {
+            audit_authz_failure(rq, auth_result, e.what());
+            throw;
+        } catch (const ss::httpd::base_exception& e) {
+            audit_authz_failure(rq, auth_result, e.what());
+            throw;
+        }
+    }
+
+    // Authenticates and authorizes the request when HTTP Basic Auth is enabled
+    // and handles audit logging. It throws on failure.
+    void handle_auth(server::request_t& rq) const {
+        rq.authn_method = config::get_authn_method(
+          rq.service().config().schema_registry_api.value(),
+          rq.req->get_listener_idx());
 
-            user = credential_t{
+        if (rq.authn_method != config::rest_authn_method::none) {
+            // Will throw 400 & 401 if auth fails
+            auto auth_result = maybe_authenticate_request(rq);
+
+            rq.user = credential_t{
               auth_result.get_username(),
               auth_result.get_password(),
               auth_result.get_sasl_mechanism()};
-        }
 
-        return user;
+            // Log successful authn
+            audit_authn(rq);
```

Comment:
I wonder if this could be reworked so that auditing success and failure are not in different functions? Would make it easier to eyeball if all the cases are covered.

Reply:
Extracted this code to a lambda now